src/Build/Utilities/Utilities.cs

```diff
@@ -323,137 +284,45 @@ internal static string CreateToolsVersionListString(IEnumerable<Toolset> toolset
         /// <summary>
         /// Figure out what ToolsVersion to use to actually build the project with. 
         /// </summary>
+        /// <remarks>
+        /// In Dev16 we dropped support for most of the logic around ToolsVersion. If an explicit ToolsVersion is specified via the
+        /// API we will use that if it's available. Otherwise in all cases the ToolsVersion is 'Current'.
+        /// </remarks>
         /// <param name="explicitToolsVersion">The user-specified ToolsVersion (through e.g. /tv: on the command line)</param>
         /// <param name="toolsVersionFromProject">The ToolsVersion from the project file</param>
         /// <param name="getToolset">Delegate used to test whether a toolset exists for a given ToolsVersion.  May be null, in which
         /// case we act as though that toolset existed.</param>
         /// <param name="defaultToolsVersion">The default ToolsVersion</param>
-        /// <param name="usingDifferentToolsVersionFromProjectFile">true if the project file specifies an explicit toolsversion but a different one is chosen</param>
+        /// <param name="usingDifferentToolsVersionFromProjectFile">true if the project file specifies an explicit ToolsVersion but a different one is chosen</param>
         /// <returns>The ToolsVersion we should use to build this project.  Should never be null.</returns>
-        internal static string GenerateToolsVersionToUse(string explicitToolsVersion, string toolsVersionFromProject, GetToolset getToolset, string defaultToolsVersion, out bool usingDifferentToolsVersionFromProjectFile)
+        internal static string GenerateToolsVersionToUse(string explicitToolsVersion, string toolsVersionFromProject,
+            GetToolset getToolset, string defaultToolsVersion, out bool usingDifferentToolsVersionFromProjectFile)
         {
-            string toolsVersionToUse = explicitToolsVersion;
-
-            // hosts may need to treat toolsversions later than the current one as the current one ... or may just 
-            // want to treat all toolsversions as though they're the current one, so give them that ability 
```

Comment:
I grepped through VS and CPS and didn't see any uses of these environment variables.